subprojects/packagefiles/arrow/meson.build

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+project(
+  'arrow',
+  'cpp',
+  'c',
+  version: '20.0.0',
+  license: 'Apache-2.0',
+  meson_version: '>=1.3.0',
+  default_options: [
+    'buildtype=release',
+    'c_std=c99',
+    'warning_level=2',
+    'cpp_std=c++17',
+  ],
+)
+
+project_args = [
+  '-Wno-unused-parameter',
+  '-Wno-array-bounds',
+  '-Wno-stringop-overflow',
+  '-Wno-aggressive-loop-optimizations',
+  '-Wno-nonnull',
+]
+
+c_compiler = meson.get_compiler('c')
+c_args = c_compiler.get_supported_arguments(project_args)
+add_project_arguments(
+  c_args,
+  language: 'c',
+)
+
+cpp_compiler = meson.get_compiler('cpp')
+cpp_args = cpp_compiler.get_supported_arguments(project_args)
+add_project_arguments(
+  cpp_args,
+  language: 'cpp',
+)
+
+git_id = get_option('git_id')
+if git_id == ''
+  git_id = run_command(
+    'git',
+    'log',
+    '-n1',
+    '--format=%H',
+    check: false,
+  ).stdout().strip()
+endif
+
+git_description = get_option('git_description')
+if git_description == ''
+  git_description = run_command(
+    'git',
+    'describe',
+    '--tags',
+    check: false,
+  ).stdout().strip()
+endif
```

Comment:
These are a bit weird in a subproject context, since they'll end up capturing the Git info of the parent project if that's being built from Git.

Reply:
Yea these are just copy/paste from the upstream project options...and even then they are mostly there for compatability with the CMake config. We can remove from this

Reply:
Ah nevermind...I don't think I can remove these as they are used as configuration data within Arrow. I'll have to raise that issue upstream as to what the expectation is in a subproject context; since Arrow is a CMake project, I am somewhat doubtful that level of detail has been thought through already

Reply:
The way the (Meson) code is written now, these will resolve to the empty string if running from an unpacked source tarball, so the empty string is a valid value.  Maybe wrap the `run_command` calls in `if not meson.is_subproject()`?